PhysicsTools/NanoAOD/plugins/LHCInfoProducer.cc

```diff
@@ -59,20 +65,27 @@ class LHCInfoProducer : public edm::global::EDProducer<edm::BeginLuminosityBlock
   void produce(edm::StreamID id, edm::Event& iEvent, const edm::EventSetup& iSetup) const override {}
 
   void globalBeginLuminosityBlockProduce(edm::LuminosityBlock& iLumi, edm::EventSetup const& iSetup) const override {
-    const auto& info = iSetup.getData(lhcinfoToken_);
+    LHCInfoCombined lhcInfoCombined(iSetup, lhcinfoPerLSToken_, lhcinfoPerFillToken_, lhcinfoToken_, useNewLHCInfo_);
     auto out = std::make_unique<nanoaod::MergeableCounterTable>();
-    out->addFloat("crossingAngle", "LHC crossing angle", info.crossingAngle());
-    out->addFloat("betaStar", "LHC beta star", info.betaStar());
-    out->addFloat("energy", "LHC beam energy", info.energy());
+    out->addFloat("crossingAngle", "LHC crossing angle", lhcInfoCombined.crossingAngle());
+    out->addFloat("betaStar", "LHC beta star", lhcInfoCombined.betaStarX);
+    out->addFloat("energy", "LHC beam energy", lhcInfoCombined.energy);
     iLumi.put(std::move(out));
   }
 
   // ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
     edm::ParameterSetDescription desc;
+    desc.add<std::string>("lhcInfoLabel", "")->setComment("label used for LHCInfo");
+    desc.add<std::string>("lhcInfoPerLSLabel", "")->setComment("label of the LHCInfoPerLS record");
+    desc.add<std::string>("lhcInfoPerFillLabel", "")->setComment("label of the LHCInfoPerFill record");
+    desc.add<bool>("useNewLHCInfo", false)->setComment("flag whether to use new LHCInfoPer* records or old LHCInfo");
```

Comment:
maybe replace "*" with "LS/Fill"

Reply:
done in last force-push [f13a480](https://github.com/cms-sw/cmssw/pull/43956/commits/f13a480eb603ac98406d219ccb307ec52a0f3b82)